routing.rst

```diff
@@ -116,57 +86,127 @@ Now you can configure the routes:
+
+    .. code-block:: yaml
+
+        # config/routes.yaml
+        api_post_show:
+            path:       /api/posts/{id}
+            controller: App\Controller\BlogApiController::show
+            methods:    GET|HEAD
+
+        api_post_edit:
+            path:       /api/posts/{id}
+            controller: App\Controller\BlogApiController::edit
+            methods:    PUT
+
+    .. code-block:: xml
+
+        <!-- config/routes.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <routes xmlns="http://symfony.com/schema/routing"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                https://symfony.com/schema/routing/routing-1.0.xsd">
 
-    However, the slash ``/`` is ignored by default in placeholder values because
-    the router uses it as separator between different placeholders.
-    To learn more about this, you can read
-    :ref:`routing/slash_in_parameter`.
+            <route id="api_post_show" path="/api/posts/{id}"
+                controller="App\Controller\BlogApiController::show"
+                methods="GET|HEAD"/>
 
-Each route also has an internal name: ``blog_list`` and ``blog_show``. These can
-be anything (as long as each is unique) and don't have any meaning yet. You'll
-use them later to :ref:`generate URLs <routing-generate>`.
+            <route id="api_post_edit" path="/api/posts/{id}"
+                controller="App\Controller\BlogApiController::edit"
+                methods="PUT"/>
+        </routes>
+
+    .. code-block:: php
+
+        // config/routes.php
+        use App\Controller\BlogApiController;
+        use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
+
+        return function (RoutingConfigurator $routes) {
+            $routes->add('api_post_show', '/api/posts/{id}')
+                ->controller([BlogApiController::class, 'show'])
+                ->methods(['GET', 'HEAD'])
+            ;
+            $routes->add('api_post_edit', '/api/posts/{id}')
+                ->controller([BlogApiController::class, 'edit'])
+                ->methods(['PUT'])
+            ;
+        };
 
-.. sidebar:: Routing in Other Formats
+.. tip::
 
-    The ``@Route`` above each method is called an *annotation*. If you'd rather
-    configure your routes in YAML, XML or PHP, that's no problem! Create a new
-    routing file (e.g. ``routes.xml``) in the ``config/`` directory and Symfony
-    will automatically use it.
+    HTML forms only support ``GET`` and ``POST`` methods. If you are calling a
+    route with a different method from an HTML form, add a hidden field called
+    ``_method`` with the method to use (e.g. ``<input type="hidden" name="_method" value="PUT" />``).
+    If you use :doc:`Symfony forms </forms>` this is done automatically for you.
 
 .. _i18n-routing:
 
-Localized Routing (i18n)
-------------------------
+Localized Routes (i18n)
+~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
This feels too soon for me. I'm looking at the in-page navigation. With this PR, it is:

```
* Creating Routes
  * Route HTTP Methods
  * Localized Routes (I18n)
  * Route Groups
* Debugging Routes
* Route Parameters
  * Parameters Validation
  * Optional Parameters
  * Expression Parameters
  * Parameter Conversion
  * Special Parameters
  * Extra Parameters
  * Slack Character in Route Parameters
* Getting Route Information
* Special Routes
  * Rendering templates
  * Redirecting to URLs and Routes
* Sub-Domain Routing
  * Testing Sub-Domain Routing
* Generating URLs
 * Generating URLs in Controllers
 * Generating URLs in ...
 * ... about 5 more of these - very nice actually
* Troubleshooting
```

Here's my proposed change: https://gist.github.com/weaverryan/372a095b9f0fabb8e27ae3308795cff5/revisions#diff-29f95331297a1d1a378b587c9181f9ea - and some explanations:

* Localized routes moves down (and because it already has no sub-sections, it is just a single-level item)
* Remove `Testing Sub-Domain Routing` header so it doesn't show up as a sub-nav item. The really shouldn't live here either...
* Put `Debugging Routes` under `Creating Routes`
* A few title renames...

Reply:
And... one additional suggestion: move "Route Groups & Prefixes" below "Route Parameters" and make it a top-level h2 (with no sub-levels). The reason is that this section (like the localized section) shows *tons* of examples with route parameters... even though we haven't talked about those yet. The "Debugging Routes" section also shows wildcards, but I think that's ok.

Reply:
All these changes make things much better. Done! Thank you.